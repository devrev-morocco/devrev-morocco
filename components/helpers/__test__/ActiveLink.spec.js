/* eslint-disable no-undef */
import { render, cleanup, screen } from '@testing-library/react';
import ActiveLink from '../ActiveLink';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

afterEach(cleanup);

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/playlist/1/devrev-1',
      pathname: '',
      query: '',
      asPath: '/playlist/1/devrev-1'
    };
  }
}));

describe('Renders ActiveLink', () => {
  it('Has Href', () => {
    const { container } = render(
      <ActiveLink href={'/playlist/1/devrev-1'} activeClassName="selected">
        <a className="foo">
          <span>playlist</span>
        </a>
      </ActiveLink>
    );

    const A_node = container.querySelector('A');
    expect(A_node).toHaveAttribute('href', '/playlist/1/devrev-1');
  });

  it('Has `selected` className', () => {
    const { container } = render(
      <ActiveLink href={'/playlist/1/devrev-1'} activeClassName="selected">
        <a className="foo">
          <span>playlist</span>
        </a>
      </ActiveLink>
    );

    const A_node = container.querySelector('A');
    expect(A_node.classList.contains('foo')).toBe(true);
    expect(A_node).toHaveClass('selected');
  });

  it('Has not `selected` className', () => {
    render(
      <ActiveLink href={'/'} activeClassName="selected">
        <a className="foo">
          <span>playlist</span>
        </a>
      </ActiveLink>
    );
    expect(
      screen.getByText('playlist').closest('a').classList.contains('foo')
    ).toBe(true);
    expect(
      screen.getByText('playlist').closest('a').classList.contains('selected')
    ).toBe(false);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ActiveLink href={'/playlist/1/devrev-1'} activeClassName="selected">
          <a className="foo">
            <span>playlist</span>
          </a>
        </ActiveLink>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
