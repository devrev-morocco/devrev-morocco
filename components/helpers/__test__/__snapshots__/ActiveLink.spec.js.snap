// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ActiveLink renders correctly 1`] = `
<a
  className="foo selected"
  href="/playlist/1/devrev-1"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <span>
    playlist
  </span>
</a>
`;
